ui/pages/confirmations/hooks/transactions/dapp-swap-comparison/useDappSwapUSDValues.ts

```diff
@@ -32,11 +32,22 @@ export function useDappSwapUSDValues({
 
   const { value: fiatRates, pending: fiatRatesPending } = useAsyncResult<
     Record<Hex, number | undefined>
-  >(() => {
+  >(async () => {
     const addresses = tokenAddresses.filter(
       (tokenAddress) => !isNativeAddress(tokenAddress),
     );
-    return fetchTokenExchangeRates('usd', addresses as Hex[], chainId);
+    const exchangeRates = await fetchTokenExchangeRates(
+      'usd',
+      addresses as Hex[],
+      chainId,
+    );
+
+    if (chainId === CHAIN_IDS.POLYGON) {
+      const nativeAddress = getNativeAssetForChainId(chainId).address;
+      exchangeRates[nativeAddress] = exchangeRates[POLYGON_NATIVE_ASSET];
```

Comment:
What is the difference between `nativeAddress` and `POLYGON_NATIVE_ASSET`?

Reply:
`0x0000000000000000000000000000000000001010` vs `0x0000000000000000000000000000000000000000`

Reply:
`0x0000000000000000000000000000000000001010` is the native address for Polygon.. are we not getting that address when swapping Pol on Polygon? 🤔